### PR TITLE
Use SDL_bool where appropriate in SDL events

### PR DESCRIPTION
**Pull request by slouken**
This involved changing button state from Uint8 to SDL_bool, and made SDL_PRESSED and SDL_RELEASED unnecessary.

Fixes https://github.com/libsdl-org/SDL/issues/10069

### CODE REVIEW

File: src/video/emscripten/SDL_emscriptenevents.c [hunk 1]
```diff
@@ -681,13 +679,11 @@ static void Emscripten_UpdatePointerFromEvent(SDL_WindowData *window_data, const
...
+            SDL_SendPenTouch(0, pen, window_data->window, false, down);
         } else if (event->button == 5) {  // eraser touch...? Not sure if this is right...
-            SDL_SendPenTouch(0, pen, window_data->window, (event->buttons & 32) ? SDL_PRESSED : SDL_RELEASED, 1);
-        } else if (event->button == 1) {
-            SDL_SendPenButton(0, pen, window_data->window, (event->buttons & 4) ? SDL_PRESSED : SDL_RELEASED, 2);
-        } else if (event->button == 2) {
-            SDL_SendPenButton(0, pen, window_data->window, (event->buttons & 2) ? SDL_PRESSED : SDL_RELEASED, 1);
+            SDL_SendPenTouch(0, pen, window_data->window, true, down);
+        } else {
+            SDL_SendPenButton(0, pen, window_data->window, event->button, down);
```
Comment (slouken) [inline src/video/emscripten/SDL_emscriptenevents.c:686]: @icculus, are these changes right? The masks and button ids in the original code seemed wrong, but...?

Comment (icculus) [reply src/video/emscripten/SDL_emscriptenevents.c:686]: Nope, it needs all those weird bitmasks:

https://developer.mozilla.org/en-US/docs/Web/API/Pointer_events#determining_button_states

**Review by icculus — CHANGES_REQUESTED** [review]
The Emscripten piece needs to change back, but the rest looks okay from 30,000 feet up.

We have a compiler assert somewhere that `sizeof (SDL_bool) == 1`, right?

**Comment by slouken** [comment]
> The Emscripten piece needs to change back, but the rest looks okay from 30,000 feet up.
> 
> We have a compiler assert somewhere that `sizeof (SDL_bool) == 1`, right?

Yep!

Fixing Emscripten now, thanks!

**Comment by sezero** [comment]
For sdl2-compat, we assume `Uint8` (`unsigned char`) and `_Bool` (or C++ `bool`) having the same memory representation, do we? (e.g. for `SDL_GetKeyboardState`)

**Comment by slouken** [comment]
> For sdl2-compat, we assume `Uint8` (`unsigned char`) and `_Bool` (or C++ `bool`) having the same memory representation, do we? (e.g. for `SDL_GetKeyboardState`)

Yep

**Pull request merged**